test/models/podcast_test.rb

```diff
@@ -17,12 +17,29 @@
     assert podcast.public_feed == podcast.default_feed
   end
 
+  it "can set a guid when id is not nil" do
+    default_guid = "79b4e97f-b349-5116-a9cf-dc479b16d518"
+    p = build(:podcast).tap(&:valid?)
+    p.default_feed.url = nil
+
+    assert_equal p.public_url, "https://f.prxu.org//feed-rss.xml"
```

Comment:
@cavis this is the problem, the public url isn't blank, but is always this exact url before the feed is persisted and the id is set, and so all guids derived from this public url are likewise identical